### PR TITLE
Create adoption register screen

### LINKED ISSUE #76: Create adoption register screen
## Overview
The screen for users register an adoption.

#30

### PR DESCRIPTION
**Pull request by rafapaivadeandrade**
closes #76
ezgif com-gif-maker (2)

Add adoption register screens(register first step, register second step)

### CODE REVIEW

**Comment by belgamo** [comment]
Can you attach some screenshots or gifs to show us the register flow?

File: packages/expo/src/modules/register/registerFirstStep/Register.tsx [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+                    {errors && <Error>{errors[field.name]}</Error>}
+                  </>
+                ))}
+                <LabelView>
+                  <LabelText>Tipo de Animal</LabelText>
+                </LabelView>
+                <ButtonsView>
+                  {animalsType.map((animal, index) => {
+                    return (
+                      <TypeButtons
```
Comment (belgamo) [inline packages/expo/src/modules/register/registerFirstStep/Register.tsx]: Please use the button from ui package. You don't need to create another one. After that you can remove the `expo-linear-gradient` from `package.json` file.

File: packages/expo/src/modules/register/registerFirstStep/styles.ts [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+          padding-left: 5px;
+          border-radius: 20px;
+        `
+      : css`
+          width: 100px;
+          text-align: center;
+          padding-top: 8px;
+        `}
+`;
+export const PortText = styled.Text<TypeProps>`
```
Comment (belgamo) [inline packages/expo/src/modules/register/registerFirstStep/styles.ts]: Please use `Typography` component from ui package to build texts. If you need to make specific changes, wrapper the base component and override its styles.

File: packages/ui/core/Button/Button.tsx [hunk 3]
```diff
@@ -128,7 +128,7 @@ const SMALL = 'small';
 type ButtonActive = boolean;
 type ButtonSize = typeof LARGE | typeof SMALL;
 type ButtonRounded = boolean;
-type ButtonText = string;
+type ButtonText = string | boolean;
```
Comment (belgamo) [inline packages/ui/core/Button/Button.tsx]: I don't think it's correct, why boolean?

File: packages/ui/core/Form/Slider.tsx [hunk 4]
```diff
@@ -13,9 +13,9 @@ const StyledSlider = styled(DefaultSlider)``;
 interface StyledSliderProps extends SliderProps {
   minimum: number;
   maximum: number;
-  value: number;
+  value?: number;
```
Comment (belgamo) [inline packages/ui/core/Form/Slider.tsx]: This shouldn't be optional.

File: packages/ui/core/Form/Slider.tsx [hunk 5]
```diff
@@ -13,9 +13,9 @@ const StyledSlider = styled(DefaultSlider)``;
...
   suffix: string;
-  onValueChange(value: number): void;
+  onValueChange?: (value: number) => void;
```
Comment (belgamo) [inline packages/ui/core/Form/Slider.tsx]: Same here, shouldn't be optional.

File: packages/expo/src/modules/register/Error.tsx [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
```
Comment (belgamo) [inline packages/expo/src/modules/register/Error.tsx]: Probably this `Error` component can be used in a lot of screens, not just in register. What do you think of putting it in the ui package?

File: packages/ui/package.json [hunk 7]
```diff
@@ -13,18 +13,21 @@
     "@react-navigation/native": "^5.0.7",
     "expo-linear-gradient": "~8.0.0",
     "expo-status-bar": "~1.0.3",
+    "formik": "^2.2.6",
```
Comment (belgamo) [inline packages/ui/package.json:18]: I think `react-hook-form` has a perfomance advantage over formik...

File: packages/expo/src/modules/register/registerFirstStep/Register.tsx [hunk 8]
```diff
@@ -0,0 +1,236 @@
...
+                {registerForm.map((field, index) => (
+                  <>
+                    <Label>{field.label}</Label>
+                    <Input
+                      key={index}
+                      placeholder={field.placeholder}
+                      onChangeText={handleChange(field.name)}
+                      value={values[field.name]}
+                    />
+                    {errors && <Error>{errors[field.name]}</Error>}
```
Comment (belgamo) [inline packages/expo/src/modules/register/registerFirstStep/Register.tsx]: I think makes sense you render the `Error` component inside `Input` and pass a condition in order to render it conditionally.

File: packages/expo/src/modules/register/Header.tsx [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+  const navigation = useNavigation();
+
+  return (
+    <HeaderView>
+      <TouchableOpacity onPress={navigation.goBack}>
+        <AntDesign name="arrowleft" size={24} color="#D4D5D4" />
+      </TouchableOpacity>
+      <TitleImageView>
+        <Typography variant={TITLE_2} type={'bold'}>
+          Cadastrar Adoção
```
Comment (belgamo) [inline packages/expo/src/modules/register/Header.tsx]: Now we have i18n in this project merged by #78. Please, translate all texts.

File: packages/expo/src/modules/register/registerFirstStep/Register.tsx [hunk 10]
```diff
@@ -0,0 +1,236 @@
...
+  race: Yup.string().required('O campo raça é obrigatório!'),
+  observations: Yup.string().required('O campo observações é obrigatório!'),
+  gender: Yup.string().required('É obrigatório escolher o sexo do animal!'),
+  animalsSize: Yup.string().required('É obrigatório escolher um tipo de animal!'),
+  animalsType: Yup.string().required('Escolha o tamanho do animal!'),
+});
+
+const LARGE = 'large';
+
+interface Values {
```
Comment (belgamo) [inline packages/expo/src/modules/register/registerFirstStep/Register.tsx:20]: We already have all the types related to adoptions declared in the graphql package. So I think you can consume from there. Maybe we can extract all of common types in a new package and then consume from it. Does this make sense @renanmav?

**Review by belgamo — CHANGES_REQUESTED** [review]

File: packages/expo/src/modules/register/registerFirstStep/Register.tsx [hunk 11]
```diff
@@ -0,0 +1,236 @@
+import React, {useState} from 'react';
+import Button from '@animavita/ui/core/Button/Button';
+import {Background} from '@animavita/ui/layout';
+import {useNavigation} from '@react-navigation/native';
+import {default as Slider} from '@animavita/ui/core/Form/Slider';
+import Input from '@animavita/ui/core/Form/Input';
+import Label from '@animavita/ui/core/Form/Label';
```
Comment (petruspierre) [inline packages/expo/src/modules/register/registerFirstStep/Register.tsx]: You can import all these `ui/core` components with just one line
`import {Input, Slider, Label, Button} from '@animavita/ui/core';`

File: packages/expo/src/modules/register/registerFirstStep/Register.tsx [hunk 12]
```diff
@@ -0,0 +1,236 @@
...
+                  <LabelText>Tipo de Animal</LabelText>
+                </LabelView>
+                <ButtonsView>
+                  {animalsType.map((animal, index) => {
+                    return (
+                      <TypeButtons
+                        key={index}
+                        start={{x: 0, y: 0.5}}
+                        end={{x: 1, y: 0.5}}
+                        colors={['#17D4B4', '#36EE8C']}
```
Comment (petruspierre) [inline packages/expo/src/modules/register/registerFirstStep/Register.tsx]: Here we could use our useTheme from `@animavita/theme`

File: packages/expo/src/modules/register/Header.tsx [hunk 13]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import {useNavigation} from '@react-navigation/native';
+import {TouchableOpacity} from 'react-native-gesture-handler';
+import {AntDesign} from '@expo/vector-icons';
+import {Typography} from '@animavita/ui/core';
+
+import {HeaderView, TitleImageView} from './registerFirstStep/styles';
```
Comment (petruspierre) [inline packages/expo/src/modules/register/Header.tsx]: Why Header styles are coming from registerFirstStep styles? The same Header is used in registerSecondStep

File: packages/expo/src/modules/register/registerFirstStep/styles.ts [hunk 14]
```diff
@@ -0,0 +1,162 @@
...
+          text-align: center;
+        `}
+`;
+
+export const Wrapper = styled.KeyboardAvoidingView`
+  margin: 0px ${px2ddp(10)}px;
+`;
+
+export const HeaderView = styled.View`
+  background-color: #f9fafc;
```
Comment (petruspierre) [inline packages/expo/src/modules/register/registerFirstStep/styles.ts]: It won't work with our dark mode. If you want to set a background-color, I would recommend you to use `useTheme` to pick the correct one according to the actual theme.

File: packages/expo/src/modules/register/registerFirstStep/styles.ts [hunk 15]
```diff
@@ -0,0 +1,162 @@
...
+export const TitleImageView = styled.View`
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  align-items: center;
+  padding-bottom: 5px;
+`;
+
+export const ScrollViewContainer = styled.ScrollView`
+  height: 420px;
```
Comment (petruspierre) [inline packages/expo/src/modules/register/registerFirstStep/styles.ts]: Why a fixed height here?

File: packages/expo/src/modules/register/registerSecondStep/RegisterSecondStep.tsx [hunk 16]
```diff
@@ -0,0 +1,152 @@
...
+                </ImageContainer>
+              );
+            }
+            return (
+              <CameraButton
+                key={index}
+                onPress={() => {
+                  handleSelectImages(index);
+                }}>
+                <Feather name="camera" size={24} color="#17D4B4" />
```
Comment (petruspierre) [inline packages/expo/src/modules/register/registerSecondStep/RegisterSecondStep.tsx]: Again, you can use our useTheme from `animavita/theme`

File: packages/expo/src/modules/register/registerSecondStep/styles.ts [hunk 17]
```diff
@@ -0,0 +1,65 @@
+import styled from 'styled-components/native';
+import {TouchableOpacity} from 'react-native-gesture-handler';
+import {px2ddp} from '@animavita/theme';
```
Comment (petruspierre) [inline packages/expo/src/modules/register/registerSecondStep/styles.ts]: You're importing px2ddp but rarely used, why?

File: packages/expo/src/modules/register/registerFirstStep/styles.ts [hunk 18]
```diff
@@ -0,0 +1,162 @@
...
+
+export const TypeButtons = styled(LinearGradient)<TypeProps>`
+  ${props =>
+    props.isAnimalChecked
+      ? css`
+          align-items: center;
+          padding-top: 4px;
+          padding-bottom: 4px;
+          padding-left: 4px;
+          padding-right: 4px;
```
Comment (petruspierre) [inline packages/expo/src/modules/register/registerFirstStep/styles.ts]: We could have just a `padding: 4px;` here (the same occurs more times in this same file)

**Pull request closed without merging**